### ISSUE TITLE
Avoid generating overloaded table valued functions with compilation errors, until they are supported

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12437]
We don't support overloaded table valued functions yet, see https://github.com/jOOQ/jOOQ/issues/4055.

Unfortunately, this currently produces compilation errors. It would be better if we simply skipped overloaded table valued functions, issuing a warning in the code generator. A lot of users don't really need the problematic function to be generated, and would be happy if the code simply compiled, instead of having to find out why it doesn't compile and how to work around it (using `<excludes/>`)

One use-case would be if someone generated all the schemas in a PostgreSQL database, including the `pg_catalog`, which contains overloaded table valued functions:
https://www.reddit.com/r/jOOQ/comments/pkma82/is_this_really_how_complicated_it_is_to_use_jooq/hc8wz6n?utm_source=share&utm_medium=web2x&context=3

It makes for a poor first impression if generated code doesn't compile.

**Comment by lukaseder** [comment]
An example query:

```sql
select
  information_schema.parameters.parameter_name,
  row_number() over (
    partition by information_schema.parameters.specific_name
    order by information_schema.parameters.ordinal_position
  ) as ordinal_position,
  information_schema.parameters.data_type,
  information_schema.parameters.character_maximum_length,
  information_schema.parameters.numeric_precision,
  information_schema.parameters.numeric_scale,
  'true' as is_nullable,
  information_schema.parameters.parameter_default as column_default,
  information_schema.parameters.udt_schema,
  information_schema.parameters.udt_name
from information_schema.routines
  join information_schema.parameters
    on (information_schema.routines.specific_catalog, information_schema.routines.specific_schema, information_schema.routines.specific_name) = (information_schema.parameters.specific_catalog, information_schema.parameters.specific_schema, information_schema.parameters.specific_name)
  join pg_catalog.pg_namespace
    on information_schema.routines.specific_schema = pg_catalog.pg_namespace.nspname
  join pg_catalog.pg_proc
    on (
      pg_catalog.pg_proc.pronamespace = pg_catalog.pg_namespace.oid
      and pg_catalog.pg_proc.proname = information_schema.routines.routine_name
    )
where (
  information_schema.routines.specific_name = 'f_overload_tvf_870220'
  and information_schema.parameters.parameter_mode <> 'IN'
  and pg_catalog.pg_proc.proretset
)
union all
select
  coalesce(information_schema.columns.column_name, 'f_overload_tvf') as column_name,
  coalesce(information_schema.columns.ordinal_position, 1) as ordinal_position,
  coalesce(information_schema.columns.data_type, information_schema.routines.data_type) as data_type,
  coalesce(information_schema.columns.character_maximum_length, information_schema.routines.character_maximum_length) as character_maximum_length,
  coalesce(information_schema.columns.numeric_precision, information_schema.routines.numeric_precision) as numeric_precision,
  coalesce(information_schema.columns.numeric_scale, information_schema.routines.numeric_scale) as numeric_scale,
  coalesce(information_schema.columns.is_nullable, 'true') as is_nullable,
  coalesce(information_schema.columns.column_default, null) as column_default,
  coalesce(information_schema.columns.udt_schema, null) as udt_schema,
  coalesce(
    information_schema.columns.udt_name,
    information_schema.routines.udt_name,
    (
      select pg_catalog.pg_type.typname
      from pg_catalog.pg_type
      where pg_catalog.pg_type.oid = pg_catalog.pg_proc.prorettype
    )
  ) as udt_name
from information_schema.routines
  left outer join information_schema.columns
    on (information_schema.routines.type_udt_catalog, information_schema.routines.type_udt_schema, information_schema.routines.type_udt_name) = (information_schema.columns.table_catalog, information_schema.columns.table_schema, information_schema.columns.table_name)
  join pg_catalog.pg_namespace
    on information_schema.routines.specific_schema = pg_catalog.pg_namespace.nspname
  join pg_catalog.pg_proc
    on (
      pg_catalog.pg_proc.pronamespace = pg_catalog.pg_namespace.oid
      and ((pg_catalog.pg_proc.proname || '_') || cast(pg_catalog.pg_proc.oid as varchar)) = information_schema.routines.specific_name
    )
where (
  information_schema.routines.specific_name = 'f_overload_tvf_870220'
  and (information_schema.routines.specific_catalog, information_schema.routines.specific_schema, information_schema.routines.specific_name) not in (
    select information_schema.parameters.specific_catalog, information_schema.parameters.specific_schema, information_schema.parameters.specific_name
    from information_schema.parameters
    where information_schema.parameters.parameter_mode = 'OUT'
  )
  and pg_catalog.pg_proc.proretset
)
order by 2;
```

Given this input schema:

```sql
CREATE FUNCTION f_overload_tvf (v1 INTEGER)
RETURNS TABLE (
    v INTEGER
)
AS $$
BEGIN
    RETURN QUERY
    SELECT *
    FROM (VALUES(v1)) t(a, b);
END
$$ LANGUAGE plpgsql;

CREATE FUNCTION f_overload_tvf (v1 TEXT)
RETURNS TABLE (
    v TEXT
)
AS $$
BEGIN
    RETURN QUERY
    SELECT *
    FROM (VALUES(v1)) t(a, b);
END
$$ LANGUAGE plpgsql;
```

It produces the same column twice:

```
|parameter_name|ordinal_position|data_type|
|--------------|----------------|---------|
|v             |1               |integer  |
|v             |2               |integer  |
```

Probably once for each function.

**Comment by lukaseder** [comment]
The first `union all` subquery produces the problem, the second one is empty in this case (see #3375)

**Comment by lukaseder** [comment]
Ah yes, we're joining on `pg_catalog.pg_proc.proname = information_schema.routines.routine_name`, instead of using the specific name 🙄

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0, 3.15.3 (https://github.com/jOOQ/jOOQ/issues/12438), and 3.14.15 (https://github.com/jOOQ/jOOQ/issues/12439)
